helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableNode.java

```diff
@@ -19,10 +19,277 @@
  * under the License.
  */
 
+import org.apache.helix.HelixException;
+import org.apache.helix.controller.dataproviders.ResourceControllerDataProvider;
+import org.apache.helix.model.ClusterConfig;
+import org.apache.helix.model.InstanceConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static java.lang.Math.max;
+
 /**
- * A placeholder before we have the implementation.
- *
- * This class represents a potential allocation of the replication.
- * Note that AssignableNode is not thread safe.
+ * This class represents a possible allocation of the replication.
+ * Note that any usage updates to the AssignableNode are not thread safe.
  */
-public class AssignableNode { }
+public class AssignableNode {
```

Comment:
Is it the same concept of "an instance which is assignable"? Dummy question, what is some instance as "unassignable"

Reply:
This name was actually defined by Neal in the POC. Sounds reasonable to me. Alternatives might be Node or Instance. That could be too generic. We'd better have a special name for the rebalancer.
To be clear, I don't have a strong opinion about the name, but Node or Instance is too generic and could be confused when we have the other Instance/Node defined somewhere else.